### Eclipse Workspace Patch 1.0
#P beeldgeluid
Index: sites/all/modules/media_browser_plus/media_browser_plus.install
===================================================================
--- sites/all/modules/media_browser_plus/media_browser_plus.install	(revision 20359)
+++ sites/all/modules/media_browser_plus/media_browser_plus.install	(working copy)
@@ -8,12 +8,37 @@
  * Implements hook_install().
  */
 function media_browser_plus_install() {
+  _media_browser_plus_ensure_field_title();
   // Make sure the standard 'field_tags' field exists.
   _media_browser_plus_ensure_field_tags();
   _media_browser_plus_ensure_field_folder();
 }
 
 /**
+ * Make sure the field_title field exists and is of the right type.
+ */
+function _media_browser_plus_ensure_field_title() {
+  $field = array(
+    'field_name' => 'field_file_title',
+    'type' => 'text',
+  	'entity_type' => 'file',
+  	'cardinality' => '1',
+    'label' => 'Title',
+    'widget' => array('type' => 'text_textfield'),
+  	'required' => 0,
+    'settings' => array(
+      'text_processing' => '0',
+    ),
+  );
+  _media_browser_plus_ensure_field($field);
+  // Ensure instance for each media bundle.
+  foreach (array_keys(media_type_get_types()) as $bundle) {
+    $field['bundle'] = $bundle;
+    _media_browser_plus_ensure_instance($field);
+  }
+}
+
+/**
  * Make sure the field_tags field exists and is of the right type.
  */
 function _media_browser_plus_ensure_field_tags() {
@@ -163,6 +188,7 @@
  * @TODO: discuss if this should be done
  */
 function media_browser_plus_uninstall() {
+  field_delete_field('field_title');
   field_delete_field('field_folder');
   field_delete_field('field_tags');
   variable_del('media_media_per_page');
Index: sites/all/modules/media_browser_plus/media_browser_plus.module
===================================================================
--- sites/all/modules/media_browser_plus/media_browser_plus.module	(revision 20362)
+++ sites/all/modules/media_browser_plus/media_browser_plus.module	(working copy)
@@ -1264,6 +1264,9 @@
     if (count($_SESSION['media-filter']['type'])) {
       $params['conditions'][] = array('property' => array('type', explode(',', $_SESSION['media-filter']['type']), 'IN'));
     }
+    if (strlen($_SESSION['media-filter']['title'])) {
+      $params['conditions'][] = array('field' => array('field_file_title', 'value', '%' . $_SESSION['media-filter']['title'] . '%', 'LIKE'));
+    }
     if (count($_SESSION['media-filter']['field_folder'])) {
       $params['conditions'][] = array('field' => array('field_folder', 'tid', $_SESSION['media-filter']['field_folder'], 'IN'));
     }
@@ -1861,7 +1864,14 @@
     $filter_active = TRUE;
   }
 
-  // Filename.
+  // Title
+  $title = '';
+  if (!empty($_SESSION['media-filter']['title'])) {
+    $title = $_SESSION['media-filter']['title'];
+    $filter_active = TRUE;
+  }
+
+  // Filename
   $filename = '';
   if (!empty($_SESSION['media-filter']['filename'])) {
     $filename = $_SESSION['media-filter']['filename'];
@@ -1914,6 +1924,12 @@
     ),
     '#attributes' => array('class' => array('filters')),
   );
+  $form['ajax_wrapper']['filters']['title'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Title'),
+    '#value' => $title,
+    '#default_value' => $title,
+  );
   $form['ajax_wrapper']['filters']['filename'] = array(
     '#type' => 'textfield',
     '#title' => t('Filename'),
@@ -1971,6 +1987,11 @@
  * Submit function for AJAX filter request.
  */
 function _media_browser_plus_library_browser_filters_submit($form, &$form_state) {
+  // Load file title filter.
+  // check_plain() should be called at display, but this module does it like this. Keeping the same
+  //  approach to prevent confusing and potential danger situations.
+  $_SESSION['media-filter']['title'] = check_plain($form_state['input']['title']);
+
   // Load filename filter.
   // check_plain() should be called at display, but this module does it like this. Keeping the same
   //  approach to prevent confusing and potential danger situations.
