### Eclipse Workspace Patch 1.0
#P beeldgeluid
Index: sites/all/modules/media_browser_plus/media_browser_plus.module
===================================================================
--- sites/all/modules/media_browser_plus/media_browser_plus.module	(revision 20339)
+++ sites/all/modules/media_browser_plus/media_browser_plus.module	(working copy)
@@ -23,6 +23,7 @@
   // Set base href for javascript requests.
   $settings = media_browser_plus_main_view_javascript_settings();
 
+  // Attach settings.
   $form['#attached']['js'][] = array(
     'type' => 'setting',
     'data' => $settings,
@@ -39,6 +40,7 @@
   }
   // Adding grid view js and css.
   $form['#attached']['js'][] = $path . '/js/media_browser_plus.admin.js';
+  $form['#attached']['js'][] = $path . '/js/media_browser_plus.library.js';
   drupal_add_library('system', 'ui.draggable');
   drupal_add_library('system', 'ui.droppable');
   // Removing options form part.
@@ -48,9 +50,26 @@
     unset($form['switch']);
   }
 
-  $form['admin'] = media_browser_plus_grid_view_form();
-  $form['media-basket'] = media_browser_plus_media_basket_form();
+  // Wrapper for ajax replace.
+  $form['ajax_wrapper'] = array(
+    '#type' => 'container',
+    '#attributes' => array(
+      'id' => 'ajax-wrapper',
+    ),
+  );
 
+  // Add filters.
+  _media_browser_plus_library_browser_filters($form, $form_state);
+
+  // Remove original media view
+  unset($form['admin']);
+
+  // Override original validation
+  $form['#validate'] = array('media_browser_plus_media_admin_validate');
+
+  $form['ajax_wrapper']['admin'] = media_browser_plus_grid_view_form();
+  $form['ajax_wrapper']['media-basket'] = media_browser_plus_media_basket_form();
+
   if (media_access('preview')) {
     $form['media-preview'] = media_browser_plus_media_preview_form();
   }
@@ -65,7 +84,7 @@
  */
 function media_browser_plus_media_admin_validate($form, &$form_state) {
   // Copy input files to values.
-  if ($form_state['values']['op'] != 'Download') {
+  if (!in_array($form_state['values']['op'], array('Download', t('Filter'), t('Reset')))) {
     $form_state['values']['files'] = $form_state['input']['files'];
     // Call the old validate.
     media_admin_validate($form, $form_state);
@@ -1817,7 +1836,7 @@
     $form['ajax_wrapper']['media-basket'] = media_browser_plus_media_basket_form(true);
   }
   return $form;
-  }
+}
 
 /**
  * Adds filters fieldset to form.
@@ -1873,6 +1892,9 @@
     //'#description' => t('Enter a part or whole filename to search for.'),
     '#attached' => array(
       'css' => array(drupal_get_path('module', 'media_browser_plus') . '/css/media_browser_plus.browser_filters.css'),
+      'library' => array(
+        array('media', 'media_browser_page')
+      ),
     ),
     '#attributes' => array('class' => array('filters')),
   );
