### Eclipse Workspace Patch 1.0
#P beeldgeluid
Index: sites/all/modules/media_youtube/js/media_youtube.js
===================================================================

--- includes/themes/media_youtube.theme.inc
+++ includes/themes/media_youtube.theme.inc
@@ -10,6 +10,15 @@
  * Preprocess function for theme('media_youtube_video').
  */
 function media_youtube_preprocess_media_youtube_video(&$variables) {
+
+  // Check which protocal to use
+  $ssl = (!empty($_SERVER['HTTPS']));
+  if ($ssl) {
+      $str_protocol = "https";
+  } else {
+      $str_protocol = "http";
+  }
+
   // Build the URL for display.
   $uri = $variables['uri'];
   $wrapper = file_stream_wrapper_get_instance_by_uri($uri);
@@ -29,10 +38,10 @@ function media_youtube_preprocess_media_youtube_video(&$variables) {
   // by an iframe, so that we can support users without Flash.
   $variables['output'] = <<<OUTPUT
     <object width="{$variables['width']}" height="{$variables['height']}">
-      <param name="movie" value="http://www.youtube.com/v/{$variables['video_id']}"></param>
+      <param name="movie" value="{$str_protocol}://www.youtube.com/v/{$variables['video_id']}"></param>
       <param name="allowFullScreen" value="{$variables['fullscreen']}"></param>
       <param name="wmode" value="transparent" />
-      <embed src="http://www.youtube.com/v/{$variables['video_id']}" type="application/x-shockwave-flash" width="{$variables['width']}" height="{$variables['height']}" allowfullscreen="{$variables['fullscreen']}"></embed>
+      <embed src="{$str_protocol}://www.youtube.com/v/{$variables['video_id']}" type="application/x-shockwave-flash" width="{$variables['width']}" height="{$variables['height']}" allowfullscreen="{$variables['fullscreen']}"></embed>
     </object>
 OUTPUT;

--- js/media_youtube.js (revision 20199)
+++ js/media_youtube.js (working copy)
@@ -52,7 +52,7 @@
   // Replace the object embed with YouTube's iframe. This isn't done by the
   // theme function because YouTube doesn't have a no-JS or no-Flash fallback.
   var video = $('<iframe class="youtube-player" type="text/html" frameborder="0"></iframe>');
-  var src = 'http://www.youtube.com/embed/' + settings.video_id;
+  var src = location.protocol + '//www.youtube.com/embed/' + settings.video_id;

   // Allow other modules to modify the video settings.
   settings.options = settings.options || {};
@@ -104,4 +104,4 @@
     .width(context.width());
 };

-})(jQuery);
\ No newline at end of file
+})(jQuery);
