### Eclipse Workspace Patch 1.0
#P Beeld en Geluid
Index: sites/all/modules/smart_paging/smart_paging.module
===================================================================
--- sites/all/modules/smart_paging/smart_paging.module	(revision 17345)
+++ sites/all/modules/smart_paging/smart_paging.module	(working copy)
@@ -47,7 +47,7 @@
  * Implements hook_form_alter()
  */
 function smart_paging_form_alter(&$form, &$form_state, $form_id) {
-  if ($form_id == 'field_ui_display_overview_form') {
+  if ($form_id == 'field_ui_display_overview_form' && $form['#entity_type'] != 'file') {
     // Modified part copy from field_ui_display_overview_form() at /modules/field_ui/field_ui.admin.inc
     // Base button element for the various formatter settings actions.
     $base_button = array(
